File: well-secure/src/main/Java/com/wellnetworks/wellsecure/service/WellGroupService.java
package com.wellnetworks.wellsecure.service;

import com.wellnetworks.wellcore.java.repository.member.WellGroupRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.stereotype.Component;

@Component
@ComponentScan("com.wellnetworks.wellcore.repository")
public class WellGroupService {
    @Autowired  // 스프링 빈 주입을 위한 어노테이션
    private WellGroupRepository wellGroupRepository;

    // 그룹 ID로 WellGroupDTO를 가져오는 메서드
//    public Optional<WellEmployeeManagerGroupDTOs> getGroupByIdx(String groupKey) {
//        return wellGroupRepository.findByemployeeManagerGroupKey(groupKey)
//                .map(WellEmployeeManagerGroupEntity::toDto);
//    }
}

File: well-secure/src/main/kotlin/com/wellnetworks/wellsecure/service/WellGroupService.kt
package com.wellnetworks.wellsecure.service

import com.wellnetworks.wellcore.domain.dto.WellGroupDTO
import com.wellnetworks.wellcore.repository.WellGroupRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.ComponentScan
import org.springframework.stereotype.Component
import java.util.*

@Component
@ComponentScan("com.wellnetworks.wellcore.repository")
class WellGroupSecurityService {
    @Autowired
    private lateinit var wellGroupRepository: WellGroupRepository

    fun getGroupByIdx(groupKey: String): Optional<WellGroupDTO> {
        val group = wellGroupRepository.findByGroupPermissionKey(groupKey)
        return group.map { it.toDto() }
    }
}